src/repository.rs

```diff
@@ -591,4 +606,23 @@ mod test {
         read.read_to_end(&mut buf).expect("read target");
         assert_eq!(buf.as_slice(), data);
     }
+
+    #[test]
+    #[should_panic]
```

Comment:
Uhm, I never tried this but it seems feasable and reasonable, I'll have a look at it.